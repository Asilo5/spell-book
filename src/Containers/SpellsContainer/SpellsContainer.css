.spells-container {
  background-image: url('https://image.freepik.com/free-photo/old-grunge-dark-textured-wooden-background-surface-old-brown-wood-texture_7182-333.jpg');
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
}