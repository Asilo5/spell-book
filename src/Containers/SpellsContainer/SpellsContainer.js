import React from 'react';
import  NavBar from '../NavBar/NavBar';
import SpellBook from '../SpellBook/SpellBook';
import SpellsForm from '../SpellsForm/SpellsForm';
import SpellsFilter from '../SpellsFilter/SpellsFilter';

import { bindActionCreators } from 'redux';
import { searchedSpell } from '../../actions';
import './SpellsContainer.css';
import { connect } from 'react-redux';


export const SpellsContainer  = ({ selectedSpell }) => {

    const completeSpells = selectedSpell.map((spell) => {
            return <SpellBook key={spell.id} {...spell}/>
        }) 

    return (
      <section className='spells-container'>
           <NavBar />
            <SpellsForm />
            <SpellsFilter />
            <div className='completed-spells'>
              {completeSpells} 
            </div>
        </section> 
    )
}

export const mapStateToProps = ({ selectedSpell, spellBook }) => ({
    selectedSpell,
    spellBook
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    searchedSpell: spell => dispatch( searchedSpell(spell) )
  }, dispatch)
)

export default connect(mapStateToProps)(SpellsContainer);