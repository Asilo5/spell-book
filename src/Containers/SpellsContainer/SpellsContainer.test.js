import React from 'react';
import { shallow } from 'enzyme';
import { SpellsContainer, mapStateToProps } from './SpellsContainer';

describe('SpellsContainer', () => {

    describe('Spells', () => {
        let wrapper;
        let mockSpells = [
            {
              id: '43453',
              spell: 'Crucio',
              type: 'curse'
            }
        ]
        beforeEach(() => {
          wrapper = shallow(<SpellsContainer 
              selectedSpell={mockSpells}
             />)
        })
    
        it('should match snapshot with all data being passed in correctly', () => {
          expect(wrapper).toMatchSnapshot();
        })
    })


    describe('mapStateToProps', () => {
        it.skip('should return an array with the spells state', () => {
            const mockState = {
                foundSpell: [
                    { _id: '5b74f38b3228320021ab628d',
                    spell: 'Lumos Solem',
                    type: 'Spell',
                    effect: 'creates a strong beam of light from the wand',
                    __v: 0 }
                   ],
                  filter: 'SEARCHED_SPELL'
              };

              const expected = {
                foundSpell: [
                    { _id: '5b74f38b3228320021ab628d',
                    spell: 'Lumos Solem',
                    type: 'Spell',
                    effect: 'creates a strong beam of light from the wand',
                    __v: 0 }
                  ]
              };

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);

        })
    })
})
