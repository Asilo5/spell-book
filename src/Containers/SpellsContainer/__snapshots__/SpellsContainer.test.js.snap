// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellsContainer Spells should match snapshot with all data being passed in correctly 1`] = `
<section
  className="spells-container"
>
  <Connect(NavBar) />
  <Connect(SpellsForm) />
  <Connect(SpellBook)
    id="43453"
    key="43453"
    spell="Crucio"
    type="curse"
  />
</section>
`;
